File: src/net/azib/photos/Entity.java
package net.azib.photos;

import java.text.SimpleDateFormat;

public class Entity {
  public String title;
  public String description;
  public String thumbUrl;
  public Long timestamp;
  public GeoLocation geo;

  public String getDescription() {
    return description != null ? description : "";
  }

  public String getTimestampISO() {
    return new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'").format(timestamp);
  }

  public Object get(String name) throws NoSuchFieldException, IllegalAccessException {
    return getClass().getField(name).get(this);
  }
}

File: src/net/azib/photos/Entity.kt
package net.azib.photos

import java.text.SimpleDateFormat
import java.util.*

open class Entity {
  var id: String? = null
  open var title: String? = null
  open var description: String? = null
    get() = field ?: ""

  var thumbUrl: String? = null
  var timestamp: Long? = null
  var geo: GeoLocation? = null

  val timestampISO: String?
    get() = timestamp?.formatISO()

  companion object {
    private val timestampFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'")
    init {
      timestampFormat.timeZone = TimeZone.getTimeZone("UTC")
    }

    private fun Long.formatISO() = synchronized(timestampFormat) {
      timestampFormat.format(this)
    }
  }
}
